agent/csi/volume_plugin_test.go

```diff
@@ -13,102 +15,160 @@ import (
 	"google.golang.org/grpc/codes"
 )
 
-func newVolumeClient(name string, nodeID string) *NodePlugin {
-	return NewNodePlugin(name, nodeID)
+func newVolumeClient(name string, nodeID string, isStaging bool) *fakeNodeClient {
+	return newFakeNodeClient(isStaging, nodeID)
```

Comment:
Same here. You need to take the fake and put it into the `NodePlugin`.

In the finished product, it will look like:

```
volumeManager -> NodePlugin -> gRPC client -> remote plugin
```

And so for testing, you do instead

```
test code -> NodePlugin -> fake client
```

whereas in this revision, you're doing

```
test code -> fake client
```